src/Reflex/Dom/Widget/Input.hs

```diff
@@ -212,6 +214,110 @@ dropdown k0 options (DropdownConfig setK attrs) = do
   dValue <- combineDyn readKey options =<< holdDyn (Just k0) (leftmost [eChange, fmap Just setK])
   return $ Dropdown dValue (attachDynWith readKey options eChange)
 
+
+data ButtonGroup t a
+   = ButtonGroup { _buttonGroup_value    :: Dynamic t (Maybe a)
+                 , _buttonGroup_select   :: Event  t (Maybe a)
+                 , _buttonGroup_element  :: El t
+                 , _buttonGroup_children :: Dynamic t (Map.Map Int (El t))
+                 }
+
+data ButtonGroupConfig t a
+   = ButtonGroupConfig { _buttonGroupConfig_parentType   :: String
+                       , _buttonGroupConfig_parentTag    :: String
+                       , _buttonGroupConfig_initialValue :: Maybe a
+                       , _buttonGroupConfig_setValue     :: Event t (Maybe a)
+                       , _buttonGroupConfig_attributes   :: Dynamic t (Map String String)
+                       }
+
+instance Reflex t => Default (ButtonGroupConfig t a) where
+  def = ButtonGroupConfig { _buttonGroupConfig_parentType   = ""
+                          , _buttonGroupConfig_parentTag    = "div"
```

Comment:
The inclusion of parentType and parentTag here feels a bit too opinionated for me.  Can we instead just assume that the caller will construct the parent element, or would this make things more error-prone?